### PR TITLE
ROADMAP: add deprecated code removal schedule

### PR DESCRIPTION
**Pull request by roman-khimov**
I think it's useful to have this publicly available.

### CODE REVIEW

File: ROADMAP.md [hunk 1]
```diff
@@ -7,5 +7,79 @@ functionality.
...
+
+## Old RPC client APIs
+
+A huge set of RPC client APIs was deprecated in versions 0.99.2 and 0.99.3
+(August-September 2022), including very frequently used ones like
+SignAndPushInvocationTx, AddNetworkFee, TransferNEP17. A new set of
+invoker/actor/unwrap/nep17/etc packages was introduced decoupling these
+functions from RPC client and simplifying typical backend code. Please refer
+to rpcclient package documentation for specific replacements for each of these
+APIs and convert your code to using them.
```
Comment (AnnaShaleva) [inline ROADMAP.md:59]: s/using/use?

File: ROADMAP.md [hunk 2]
```diff
@@ -7,5 +7,79 @@ functionality.
...
+version 0.97.3. Since then we have deprecated Magic and StateRootInHeader
+fields in the Version structure that allows to decode replies from pre-0.97.3
+servers while RPC server implementation populates both old and new fields for
+compatibility with pre-0.97.3 clients.
+
+Version 0.97.3 was released in October 2021 and can't really be used today on
+public networks, we expect at least 0.99.0+ to be used in production (0.99.2+
+for C# 3.4.0 compatibility). Therefore these old fields are scheduled to be
+removed in version 0.100.0 both client-side and server-side. If any of your
+code uses them, just use the Protocol section with the same data.
```
Comment (AnnaShaleva) [inline ROADMAP.md:39]: There's a `"protocol" section` a few lines above.

File: ROADMAP.md [hunk 3]
```diff
@@ -7,5 +7,79 @@ functionality.
...
+to rpcclient package documentation for specific replacements for each of these
+APIs and convert your code to using them.
+
+While a lot of the code is already converted to new APIs, old ones still can
+be used in some code not known to us. Therefore we will remove old APIs not
+earlier than February-March 2023, with 0.101.0 or 0.102.0 release.
+
+## util.FromAddress smart contract helper
+
+`util` smart contract library has a FromAddress function that is one of oldest
```
Comment (AnnaShaleva) [inline ROADMAP.md:67]: s/of oldest/of the oldest

**Review by AnnaShaleva — COMMENTED** [review]
Maybe we'd like to add a couple of words about deprecated Prometheus RPC server metrics in case if someone uses them?

File: ROADMAP.md (hunk 1 above)
Comment (roman-khimov) [reply ROADMAP.md:59]: Both seem to be fine to me.

**Comment by roman-khimov** [comment]
> Maybe we'd like to add a couple of words about deprecated Prometheus RPC server metrics in case if someone uses them?

Right! That's what I wanted to add also, but was distracted.

File: ROADMAP.md (hunk 3 above)
Comment (roman-khimov) [reply ROADMAP.md]: Fixed.

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**